lib/JSON/Validator.pm

```diff
@@ -187,7 +187,7 @@ sub validate {
   local $self->{seen}        = {};
   local $self->{temp_schema} = [];                            # make sure random-errors.t does not fail
   my @errors = $self->_validate($_[1], '', $schema);
-  return @errors;
+  return sort { $a->path() cmp $b->path() } @errors;
```

Comment:
Note that the behaviour of `sort` in scalar context is undefined.
If anyone is calling `validate` in scalar context to get the number of errors, with perl 5.30.3 for example it will return `undef`.

Reply:
Ooh good catch @perlpunk, that's nasty